zip-0312.rst

```diff
@@ -1,7 +1,466 @@
+------------
+
+FROST(Jubjub, BLAKE2b-512)
+'''''''''''''''''''''''''''''
+
+This ciphersuite uses Jubjub for the Group and BLAKE2b-512 for the Hash function `H`
+meant to produce signatures indistinguishable from RedJubjub Sapling Spend
+Authorization Signatures as specified in [#protocol-concretespendauthsig]_.
+
+- Group: Jubjub [#protocol-jubjub]_
+
+  - Order: 6554484396890773809930967563523245729705921265872317281365359162392183254199 (see [#protocol-jubjub]_)
+  - Identity: as defined in [#protocol-jubjub]_
+  - RandomScalar(): Implemented by returning a uniformly random Scalar in the range
+    \[0, `G.Order()` - 1\]. Refer to {{frost-randomscalar}} for implementation guidance.
+  - SerializeElement(P): Implemented as :math:`\mathsf{repr}_\mathbb{J}(P)` as defined in [#protocol-jubjub]_
+  - DeserializeElement(P): Implemented as :math:`\mathsf{abst}_\mathbb{J}(P)` as defined in [#protocol-jubjub]_,
+    failing if :math:`\bot` is returned. Additionally, this function validates that the resulting
+    element is not the group identity element, returning an error if the check fails.
+  - SerializeScalar: Implemented by outputting the little-endian 32-byte encoding
+    of the Scalar value.
+  - DeserializeScalar: Implemented by attempting to deserialize a Scalar from a
+    little-endian 32-byte string. This function can fail if the input does not
+    represent a Scalar in the range \[0, `G.Order()` - 1\].
+
+- Hash (`H`): BLAKE2b-512 [#BLAKE]_ (BLAKE2b with 512-bit output and 16-byte personalization string),
+  and Nh = 64.
+
+  - H1(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubR", m), interpreting
+    the 64 bytes as a little-endian integer, and reducing the resulting integer
+    modulo L = 6554484396890773809930967563523245729705921265872317281365359162392183254199.
+  - H2(m): Implemented by computing BLAKE2b-512("Zcash_RedJubjubH", m), interpreting
+    the 64 bytes as a little-endian integer, and reducing the resulting integer
+    modulo L = 6554484396890773809930967563523245729705921265872317281365359162392183254199.
+    (This is equivalent to :math:`\mathsf{H}^\circledast(m)`, as defined in
+    [#protocol-concretereddsa]_ parametrized with [#protocol-jubjub]_.)
+  - H3(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubN", m), interpreting
+    the 64 bytes as a little-endian integer, and reducing the resulting integer
+    modulo L = 6554484396890773809930967563523245729705921265872317281365359162392183254199.
+  - H4(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubM", m)
+  - H5(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubC", m)
+
+
+FROST(Pallas, BLAKE2b-512)
+'''''''''''''''''''''''''''''
+
+This ciphersuite uses Pallas for the Group and BLAKE2b-512 for the Hash function `H`
+meant to produce signatures indistinguishable from RedPallas Orchard Spend
+Authorization Signatures as specified in [#protocol-concretespendauthsig]_.
+
+- Group: Pallas [#protocol-pallasandvesta]_
```

Comment:
The FROST spec's notion of a group includes a base point, which in this case is $\mathcal{G}^{Orchard}$ defined in 5.4.7.1.

Reply:
Done in f72d8cd